src/game_battlealgorithm.cpp

```diff
@@ -1031,7 +1036,7 @@ bool Game_BattleAlgorithm::Normal::Execute() {
 				}
 			}
 			if (Game_Battle::GetBattleCondition() == lcf::rpg::System::BattleCondition_back) {
-				if (source->GetType() == Game_Battler::Type_Ally && static_cast<Game_Actor*>(source)->GetBattleRow() == Game_Actor::RowType::RowType_front || target->GetType() == Game_Battler::Type_Ally && static_cast<Game_Actor*>(target)->GetBattleRow() == Game_Actor::RowType::RowType_front) {
+				if (target->GetType() == Game_Battler::Type_Enemy || target->GetType() == Game_Battler::Type_Ally && static_cast<Game_Actor*>(target)->GetBattleRow() == Game_Actor::RowType::RowType_front) {
```

Comment:
This `Type_Enemy` thing looks like a bug? Also lets smash all this into one commit so it's easier to review

Reply:
The back attack condition is a special case. If the player attacks an enemy the damage is always reduced by 25% regardless of row. If the player is in the front row he gets the 25% damage bonus but it still means the the damage is reduced by 6.25%. That were the results from my tests with RPG_RT.

Reply:
You're right. I checked again.

As mentioned in the issue, there are 2 normal attack code paths. One appears to be taken for normal, initiative, and surround. The other for back and pincer.

The first code path only does row adjustements on the attacker / defender if it's an actor. The second one does not.

Enemies are always in the front row. So for back you always miss more often and hit for less. I need to verify this more, it looks like RPG_RT bugs because this behavior does not make a lot of sense.

Also we need to verify when actors attack other attacks via confuse status.

Reply:
I have tested the confused status cases and in this cases the modifiers are applied as well. I only had to check in back attack condition if source and target is an ally because the 25% damage bonus is not applied if an ally attacks an other ally or himself due to confusion.

Reply:
Also lets put parens around the second condition to make the logic more clear

`target->GetType() == Game_Battler::Type_Enemy || (target->GetType() == Game_Battler::Type_Ally && static_cast<Game_Actor*>(target)->GetBattleRow() == Game_Actor::RowType::RowType_front)`

Reply:
Parenthesis have been added.